src/Authorization/Services/Implementation/DelegationContextHandler.cs

```diff
@@ -43,28 +45,87 @@ public DelegationContextHandler(IInstanceMetadataRepository policyInformationRep
         /// <summary>
         /// Updates needed subject information for the Context Request for a specific delegation
         /// </summary>
-        /// <param name="request">The original Xacml Context Request</param>
-        /// <param name="subjects">The list of PartyIds to be added as subject attributes</param>
-        public void Enrich(XacmlContextRequest request, List<int> subjects)
+        /// <param name="requestSubjectAttributes">The current collection of subject attributes on the request to be enriched</param>
+        /// <param name="isInstanceAccessRequest">Whether the request is for a specific instance, which needs additional uuid information</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/></param>
+        public async Task EnrichRequestSubjectAttributes(XacmlContextAttributes requestSubjectAttributes, bool isInstanceAccessRequest, CancellationToken cancellationToken)
         {
-            if (subjects?.Count == 0)
+            int subjectUserId = GetSubjectUserId(requestSubjectAttributes);
+            if (subjectUserId > 0)
```

Comment:
Just a comment for later discussion. 
When we update the cookie with useruuid the standard pep should maybe start sending that. 
Should it then send both? Should this code check for useruid.

Reply:
Jo, det må utvidelser til her for å støtte uuid subjects etterhvert